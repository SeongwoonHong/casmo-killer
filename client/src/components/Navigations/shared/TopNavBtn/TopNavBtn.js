import React from 'react';
import PropTypes from 'prop-types';

import './TopNavBtn.scss';

const TopNavBtn = ({
  className, classPrefix, icon, onClick, children
}) => {

  return (
    <button
      type="button"
      className={ `Top-nav-btn ${classPrefix}__btn ${className}`}
      onClick={ () => onClick() }>
      {
        children ||
        <i className={ `Top-nav-btn__icons material-icons ${classPrefix}__icon`}>
          { icon }
        </i>
      }
    </button>
  );

};

TopNavBtn.propTypes = {
  className: PropTypes.string,
  classPrefix: PropTypes.string,
  icon: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.any,
};

TopNavBtn.defaultProps = {
  className: '',
  classPrefix: '',
  icon: '',
  onClick: () => {},
  children: '',
};

export default TopNavBtn;
