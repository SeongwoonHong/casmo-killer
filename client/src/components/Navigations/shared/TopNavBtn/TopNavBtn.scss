.Top-nav-btn {
    @include rectanglify($top-nav-size);
    background-color: transparent;
    border: none;
    color: palette(Grey, 100);
    &:focus {
        outline: none;
    }
    &__icons {
        font-size: 24px;
        height: 100%;
        line-height: $top-nav-size;
    }
}
