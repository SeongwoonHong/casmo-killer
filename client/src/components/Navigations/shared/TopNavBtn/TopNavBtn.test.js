import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import TopNavBtn from './index';

describe('<TopNavBtn />', () => {

  let component = null;

  beforeEach(() => {
    component = shallow(<TopNavBtn />);
  });

  it('renders the component consistently', () => {

    expect(component).toMatchSnapshot();

  });

  it('fires onClick event handler', () => {

    const onClick = sinon.spy();

    const wrapper = shallow(
      <TopNavBtn onClick={ onClick } />
    );

    wrapper.find('button').simulate('click');

    expect(onClick.calledOnce).toEqual(true);

  });

});
