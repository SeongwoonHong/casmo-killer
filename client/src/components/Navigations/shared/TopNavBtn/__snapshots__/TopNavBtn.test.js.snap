// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNavBtn /> renders the component consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopNavBtn
    className=""
    classPrefix=""
    icon=""
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <i
        className="Top-nav-btn__icons material-icons __icon"
/>,
      "className": "Top-nav-btn __btn ",
      "onClick": [Function],
      "type": "button",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "className": "Top-nav-btn__icons material-icons __icon",
      },
      "ref": null,
      "rendered": "",
      "type": "i",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <i
          className="Top-nav-btn__icons material-icons __icon"
/>,
        "className": "Top-nav-btn __btn ",
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "Top-nav-btn__icons material-icons __icon",
        },
        "ref": null,
        "rendered": "",
        "type": "i",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
